scripts.py

```diff
@@ -300,37 +337,35 @@ class WorkProcessingScript(IdentifierInputScript):
     name = "Work processing script"
 
     def __init__(self, force=False, batch_size=10):
-
-        identifiers_or_source = self.parse_identifiers_or_data_source()
+        args = self.parse_command_line(self._db)
+        identifier_type = args.identifier_type
         self.batch_size = batch_size
-        self.query = self.make_query(identifiers_or_source)
+        self.query = self.make_query(identifier_type, args.identifiers)
         self.force = force
 
-    def make_query(self, identifiers):
+    def make_query(self, identifier_type, identifiers):
         query = self._db.query(Work)
-        if not identifiers:
-            self.log.info(
-                "Processing all %d works.", query.count()
-            )
-        elif isinstance(identifiers, DataSource):
-            # Find all works from the given data source.
-            query = query.join(Edition).filter(
-                Edition.data_source==identifiers
-            )
+        if identifiers or identifier_type:
+            query = query.join(Work.license_pools)
+        if identifiers:
+            query = query.join(LicensePool.identifier)
+
+        if identifiers:
             self.log.info(
-                "Processing %d works from %s", query.count(),
-                identifiers.name
+                'Restricted to %d specific identifiers.' % len(identifiers)
             )
-        else:
-            # Find works with specific identifiers.
-            query = query.join(Edition).filter(
-                    Edition.primary_identifier_id.in_(
-                        [x.id for x in identifiers]
-                    )
+            query = query.filter(
+                LicensePool.identifier_id.in_([x.id for x in identifiers])
             )
+        if identifier_type:
             self.log.info(
-                "Processing %d specific works." % query.count()
+                'Restricted to identifier type "%s".' % identifier_type
             )
+            query = query.filter(Identifier.type==identifier_type)
```

Comment:
I think it won't work. I think I did it that way because I was planning on filtering LicensePool.data_source_id but then I got rid of the data source stuff altogether.